File: icl_backend/src/main/java/es/uoc/icl/domain/PerfilNegocio.java
package es.uoc.icl.domain;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PerfilNegocio {
	private Integer id;
    private String nombre;
    private String direccion;
    private String cif;
    private String email;
    private BigDecimal valorTotal;
    private TipoNegocio tipo;
    private BigDecimal totalCanjeado;
    
    public PerfilNegocio(Negocio negocio) {
    	this.id = negocio.getId();
    	this.nombre = negocio.getNombre();
    	this.direccion = negocio.getDireccion();
    	this.cif = negocio.getCif();
    	this.email = negocio.getEmail();
    	this.valorTotal = negocio.getValorTotal();
    	this.tipo = negocio.getTipo();
    	this.totalCanjeado = negocio.getTotalCanjeado();
    }
}

File: icl_frontend/src/app/models/perfil-negocio.ts

import { TipoNegocio } from "./tipo-negocio";

export class PerfilNegocio {
  id: string;
  nombre: string;
  direccion: string;
  cif: string;
  email: string;
  valorTotal: number;
  tipo: TipoNegocio;
  totalCanjeado: number;

  constructor(
    id: string = '',
    nombre: string = '',
    direccion: string = '',
    cif: string = '',
    email: string = '',
    valorTotal: number = 0,
    tipo: TipoNegocio = new TipoNegocio,
    totalCanjeado: number = 0
  ) {
    this.id = id;
    this.nombre = nombre;
    this.direccion = direccion;
    this.cif = cif;
    this.email = email;
    this.valorTotal = valorTotal;
    this.tipo = tipo;
    this.totalCanjeado = totalCanjeado;
  }
}